2017/JULY/JULY17/EXPTREE/EXPTREE.py

```diff
@@ -0,0 +1,57 @@
+#__BY:sirjan13__
+
+
+def gcd(n, m):
+    if n == 0:
+        return m
+    return gcd(m % n, n)
+
+
+t1, t2 = 0, 0
+
+
+def egcd(a, b):
+    x, y, u, v = 0, 1, 1, 0
+    while a != 0:
+        q, r = b//a, b % a
+        m, n = x-u*q, y-v*q
+        b, a, x, y, u, v = a, r, u, v, m, n
+    gc = b
+    return gc, x, y
+
+
+def mi(a, m):
+    g = egcd(a, m)
+    ans = (g[1] % m + m) % m
+    return ans
+
+
+mod1 = 10**9+7
+mod2 = mod1 + 2
+for cppp in range(int(raw_input())):
```

Comment:
The variable names in codechef solution were much better @sirjan13 😆

Reply:
haha :100: